asdf/tags/core/ndarray.py

```diff
@@ -408,32 +411,56 @@ def reserve_blocks(cls, data, ctx):
 
     @classmethod
     def to_tree(cls, data, ctx):
+        # The ndarray-1.0.0 schema does not permit 0 valued strides.
+        # Perhaps we'll want to allow this someday, to efficiently
+        # represent an array of all the same value.
         if any(stride == 0 for stride in data.strides):
             data = np.ascontiguousarray(data)
 
-        base = util.get_array_base(data)
         shape = data.shape
-        dtype = data.dtype
-        offset = data.ctypes.data - base.ctypes.data
 
-        if data.flags.c_contiguous:
+        block = ctx.blocks.find_or_create_block_for_array(data, ctx)
+
+        if block.array_storage == "fits":
+            # Views over arrays stored in FITS files have some idiosyncracies.
+            # astropy.io.fits always writes arrays C-contiguous with big-endian
+            # byte order, whereas asdf preserves the "contiguousity" and byte order
+            # of the base array.
+            if (block.data.shape != data.shape or
+                block.data.dtype.itemsize != data.dtype.itemsize or
+                block.data.ctypes.data != data.ctypes.data or
+                block.data.strides != data.strides):
+                raise ValueError("Views over FITS arrays must have identical memory layout")
```

Comment:
For a user who doesn't know what `memory layout` means, it might be good to say something like "Quit trying to serialize a numpy view!"  in the error message, with a link to the numpy docs about the differences between arrays and views of an array.

https://numpy.org/doc/stable/glossary.html#term-view

Reply:
Good point, that message is a bit cryptic.  I'll rework it.